vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -259,40 +260,21 @@ func configFromEnv() (cfg Config, ok bool) {
 	return
 }
 
-func createKubernetesClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
-	klog.Info("Creating kubernetes API client.")
-
-	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
-	if err != nil {
-		return nil, err
-	}
-
-	client, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	v, err := client.Discovery().ServerVersion()
-	if err != nil {
-		return nil, err
-	}
-
-	klog.Infof("Kubernetes API client created, server version %s", fmt.Sprintf("v%v.%v", v.Major, v.Minor))
-	return client, nil
+// SetInformers initializes secret lister
+func (os *OpenStack) SetInformers(informerFactory informers.SharedInformerFactory) {
+	os.secretLister = informerFactory.Core().V1().Secrets().Lister()
 }
 
 // setConfigFromSecret allows setting up the config from k8s secret
 func (os *OpenStack) setConfigFromSecret() error {
+	if os.secretLister == nil {
+		return fmt.Errorf("secret lister is not initialized")
+	}
+
 	secretName := os.secretName
 	secretNamespace := os.secretNamespace
-	kubeconfigPath := os.kubeconfigPath
-
-	k8sClient, err := createKubernetesClient(kubeconfigPath)
-	if err != nil {
-		return fmt.Errorf("failed to get kubernetes client: %v", err)
-	}
 
-	secret, err := k8sClient.CoreV1().Secrets(secretNamespace).Get(secretName, metav1.GetOptions{})
+	secret, err := os.secretLister.Secrets(secretNamespace).Get(secretName)
```

Comment:
I like this much more. Only concert: we ignore the kubeconfig now passed in `os`. I guess that's ok?

Reply:
Is there some sensible fallback solution?

Reply:
Yeah, I did it for consistency. If we remove this parameter from the config, but leave it in the file, there will be an error.
I think that the best option is to comment in the code that this parameter is deprecated and should not used.